Userland/Libraries/LibWeb/Fetch/Fetching/Fetching.cpp

```diff
@@ -1108,8 +1108,24 @@ WebIDL::ExceptionOr<JS::NonnullGCPtr<PendingResponse>> http_redirect_fetch(JS::R
 
     // FIXME: 19. Invoke set request’s referrer policy on redirect on request and actualResponse.
 
-    // 20. Return the result of running main fetch given fetchParams and true.
-    return TRY(main_fetch(realm, fetch_params, Recursive::Yes)).release_value();
+    // 20. Let recursive be true.
+    auto recursive = Recursive::Yes;
+
+    // 21. If request’s redirect mode is "manual", then:
+    if (request->redirect_mode() == Infrastructure::Request::RedirectMode::Manual) {
+        // 1. Assert: request’s mode is "navigate".
+        VERIFY(request->mode() == Infrastructure::Request::Mode::Navigate);
+
+        // 2. Set recursive to false.
+        recursive = Recursive::No;
+    }
+
+    // 20. Return the result of running main fetch given fetchParams and recursive.
+    if (auto result = TRY(main_fetch(realm, fetch_params, recursive)); result.has_value()) {
+        return result.release_value();
+    }
+
+    return nullptr;
```

Comment:
Please explain this, it's not clear why the previous assumption of a non-empty optional is no longer true.